tests/datasets/test_upload/test_dataset_upload.py

```diff
@@ -0,0 +1,368 @@
+
+import csv
+import pytest
+import pandas as pd
+from io import StringIO, BytesIO
+from django.core.exceptions import ValidationError
+from django.core.files.uploadedfile import SimpleUploadedFile
+
+from tests.datasets.factories import (
+    GeographyFactory, GeographyHierarchyFactory, DatasetFactory,
+    DatasetFileFactory, IndicatorFactory, GroupFactory, DatasetDataFactory
+)
+from tests.profile.factories import ProfileFactory
+from wazimap_ng.datasets.models import DatasetData, Group
+from wazimap_ng.datasets.tasks import process_uploaded_file, indicator_data_extraction
+
+from .upload_data import (
+    dataset_upload_data, default_result_set, required_headers
```

Comment:
why not mark these as fixtures?

Reply:
Checking how to mark fixtures in your tests @adieyal

Reply:
I haven't marked imports as fixtures but used the imported values in params of fixtures to create test_data and expected value.
So now I do not have to apply parametrize on test.
Please let me know how can I  improve further or this is accurate approach for pytest :)
Thanks for the review